crates/bevy_ecs/src/component.rs

```diff
@@ -347,6 +347,13 @@ impl Components {
 
         ComponentId(*index)
     }
+
+    pub(crate) fn non_send_components(&'_ self) -> impl Iterator<Item = ComponentId> + '_ {
```

Comment:
Interesting. This will be useful if we want to relax the bounds on `Component` later too.